x-pack/filebeat/input/httpjson/encoding.go

```diff
@@ -146,7 +146,7 @@ func decodeAsCSV(p []byte, dst *response) error {
 	// values to keys in the event
 	header, err := r.Read()
 	if err != nil {
-		if err == io.EOF {
+		if err == io.EOF { //nolint:errorlint // csv.Reader never wraps io.EOF.
```

Comment:
While this is true, maybe we should go ahead and change these two over to to errors.Is() anyways?

Reply:
I'm not convinced. Errors is requires at least one call to reflectlite in all cases in order to check whether the error value is comparable. I think that if we know that the error is always unwrapped (for example it comes from a stdlib function that always did, like in the case of all uses of `io.EOF`) then it would be better to use the old convention; also for readability (though this is harmed by nolint spam).

I am happy to go over and change them all if you feel strongly about it though.

Reply:
I don't feel strongly about. I just haven't accepted that we'll have so many of these`//nolint` statements 😄.

Reply:
> I just haven't accepted that we'll have so many of these`//nolint` statements 😄.

Indeed. This is a consequence of the approach that golangci-lint takes to linting; it tries to cover too much and it does so with sometimes less than careful linters.